DE: Committee Scraper Updated and MA:Vote Scraper Updated and HI: "carried over" bills bug solved

hiteshgarg14 opened the pull request:
#1445 
Scraper is now working for `senate`, `house` and `joint` sections.

hiteshgarg14 commented:
#1162 
Vote Scraper Updated.

hiteshgarg14 commented:
#1196 
HI: "carried over" bills bug solved.

jamesturk commented on openstates/de/committees.py:
```diff
@@ -8,104 +8,87 @@ class DECommitteeScraper(CommitteeScraper,LXMLMixin):
+            self.save_committee(committee)
+
 
     def scrape_joint_committees(self,term,session):
```
is this just a copy pasted version of the other committee scraper?  if so let's go ahead & combine these

hiteshgarg14 replied:
Actually, in URLs variable, I added 'Joint' 
```
urls = {
            'upper': 'http://legis.delaware.gov/json/Committees/' +
                     'GetCommitteesByTypeId?assemblyId=%s&committeeTypeId=1',
            'lower': 'http://legis.delaware.gov/json/Committees/' +
                     'GetCommitteesByTypeId?assemblyId=%s&committeeTypeId=2',
           'joint': 'http://legis.delaware.gov/json/Committees/' + 
              'GetCommitteesByTypeId?assemblyId=%s&committeeTypeId=3'
        }
```

`url = urls[chamber] % (session,)`

 But on running scraper, only `upper` and `lower` were values of `chamber` variable. I tried it by adding `joint` in  __init__.py file's `chamber` dictionary but nothing happened. that's why I used `scrape_joint_committees` method. This is the same technique which was used before to scrape the `joint` committees.

jamesturk commented on openstates/de/committees.py:
```diff
@@ -8,104 +8,87 @@ class DECommitteeScraper(CommitteeScraper,LXMLMixin):
+                        'committeeId='+ str(comm_id)
+            comm_page = lxml.html.fromstring(self.get(comm_url).text)
+            #all members including chair_man and vice_chair
+            members = comm_page.xpath("//section[@class='section-short']/" + \
```
these line continuations aren't needed since you're inside parens

jamesturk commented on openstates/ma/votes.py:
```diff
@@ -1,209 +1,131 @@
+
+        c = 0
+        for i in final:
+        	if c == 1 and i != '  NAYS.' and i != '':
```
please use 4-spaces for indentation

hiteshgarg14 commented:
Hello :) I have made changes and fixed the committee scraper bug.

When in JSON data if we get `ChairName` field as `null` then it will be saved as `No`. similarly for `ViceChairName` field and in members name, I have fixed the bug. I was unable to check it locally as I have converted DE legislators.py to pupa. please have a look and sorry for one PR instead of 3 PR's.

jamesturk commented:
there's now partial pupa work on this branch, you will  need to separate this out into multiple branches so that we can pull in parts of it without the pupa bits.  it might be best to separate it into multiple PRs at that point too

hiteshgarg14 commented:
Opened 3 separate PR's with bugs fixed. Please have a look. 
Closing this PR.
